openedx/core/djangoapps/user_api/views.py

```diff
@@ -255,13 +255,27 @@ def get(self, request):
             'ALLOW_REGISTRATION_FORM_FIELD_OVERRIDE',
             settings.FEATURES.get('ALLOW_REGISTRATION_FORM_FIELD_OVERRIDE', False)
         ):
-            form_desc.override_field_properties(
-                'country',
-                default=configuration_helpers.get_value(
-                    'REGISTRATION_COUNTRY_FIELD_DEFAULT',
-                    settings.FEATURES.get('REGISTRATION_COUNTRY_FIELD_DEFAULT', ''))
+
+            default_fields_config = {
+                "config" : [
+                    {'field': 'country', 'default': ''},
+                    {'field': 'gender', 'default': ''},
+                    {'field': 'year_of_birth', 'default': ''},
+                    {'field': 'level_of_education', 'default': ''}
+                ]
+            }
+
+            additional_fields = configuration_helpers.get_value(
+                'REGISTRATION_FIELD_DEFAULTS',
+                settings.FEATURES.get('REGISTRATION_FIELD_DEFAULTS', default_fields_config)
```

Comment:
Thanks a lot for this change! It looks much better!

One thing, as similar to the comment above: any particular reason we need a `config` field? And would it not be better to let the defaults get set in configuration rather than in the code directly like this? I.e., you could add this variable somewhere in `lms/envs/common.py` or similar, I suppose.

Reply:
@UmanShahzad While we were developing this feedback, we faced a problem passing directly a list in the feature flag. We could not iterate it due to site_configuration expects a dict, so we fix it passing a dictionary to the feature flag and inside this dict, we pass a key with the list that we need, and finally, we could iterate it.